code/game/objects/items/weapons/RCD.dm

```diff
@@ -131,6 +131,61 @@
 	GLOB.rcd_list -= src
 	return ..()
 
+/obj/item/rcd/suicide_act(mob/living/user)
+	user.Immobilize(10 SECONDS) // You cannot move.
+	flags |= NODROP				// You cannot drop. You commit to die.
+	var/turf/suicide_tile = get_turf(src)
+	if(mode == MODE_DECON && checkResource(5, user))	// Same cost as deconstructing a wall.
+		user.visible_message("<span class='suicide'>[user] points [src] at [user.p_their()] chest and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(loc, 'sound/machines/click.ogg', 50, TRUE)
+		to_chat(user, "<span class='notice'>Deconstructing User...</span>")
+		var/obj/effect/temp_visual/rcd_effect/reverse/suicide_A = new(suicide_tile)
+		if(!do_after(user, 5 SECONDS))					// Deconstruction of most structures takes 5 seconds, wait for the animation to finish.
+			return
+		QDEL_NULL(suicide_A)
+		if(!(user.l_hand == src) && !(user.r_hand == src))	// Do not commit die if the RCD isn't in your hands.
+			return SHAME								// If this triggers, someone probably chopped off your hand mid-suicide for some reason.
+		useResource(5, user)	// Consume ammo.
+		user.visible_message("<span class='suicide'>[user] deconstructs [user.p_they()]self with [src]!</span>")
+		playsound(loc, usesound, 50, TRUE)
+		for(var/obj/item/W in user)	// Do not delete all their stuff.
+			user.unEquip(W)			// Dump everything on the floor instead.
+		flags &= ~NODROP			// NODROP must be removed so the RCD doesn't get dusted along with them. Having this come after the unequipping puts the RCD on top of the pile of stuff (held items fall to the floor when dusting).
+		user.dust()
+		return OBLITERATION
+
+	user.visible_message("<span class='suicide'>[user] puts the barrel of [src] into [user.p_their()] mouth and pulls the trigger. It looks like [user.p_theyre()] trying to commit suicide!</span>")
```

Comment:
This shouldnt be before the first suicide method?

Reply:
First suicide method has its own text where they point the RCD at their chest to deconstruct themselves.

All other suicide methods involving construction have the RCD stuck in the mouth so the constructed object appears inside the user (allowing for them to explode from having a structure suddenly appearing inside their body).